### ISSUE TITLE
Error: rpc error: code = Unavailable desc = transport is closing

### ISSUE THREAD

**Issue by nunofernandes** [terraform-aws-modules/terraform-aws-ec2-instance#128]
Hi,

I started to having this error when applying
```
Error: rpc error: code = Unavailable desc = transport is closing
```

It's for a m5 instance and the change reports
```
...
      + credit_specification {}
...
```

Apparently it due to this line https://github.com/terraform-aws-modules/terraform-aws-ec2-instance/blob/master/main.tf#L87

Using terraform 0.12.7. Here is the trace:
```
module.nginx-b.aws_instance.this[0]: Modifying... [id=i-012934007208c7aee]
2019/09/04 14:03:02 [DEBUG] module.nginx-b.aws_instance.this[0]: applying the planned Update change
2019/09/04 14:03:02 [TRACE] GRPCProvider: ApplyResourceChange
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: panic: interface conversion: interface {} is nil, not map[string]interface {}
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: 
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: goroutine 93 [running]:
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsInstanceUpdate(0xc000297c00, 0x3d6a320, 0xc000b94480, 0x0, 0x0)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_instance.go:1198 +0x4739
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0003ad200, 0xc000879d60, 0xc0007e6da0, 0x3d6a320, 0xc000b94480, 0xc000792501, 0x9413490, 0x426b080)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:293 +0x264
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc00066ca80, 0xc000a43a58, 0xc000879d60, 0xc0007e6da0, 0xc000768988, 0xc00014b238, 0x426d840)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:285 +0x18f
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00014ac50, 0x5519700, 0xc000732540, 0xc000a6cea0, 0xc00014ac50, 0xc000732540, 0xc00084fbd0)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/plugin/grpc_provider.go:885 +0x894
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x4b43f40, 0xc00014ac50, 0x5519700, 0xc000732540, 0xc000879ae0, 0x0, 0x5519700, 0xc000732540, 0xc000735500, 0x1380)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/internal/tfplugin5/tfplugin5.pb.go:3217 +0x23e
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005b8a80, 0x553bb20, 0xc0005b9200, 0xc000533200, 0xc0006618c0, 0x9413b00, 0x0, 0x0, 0x0)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:966 +0x470
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).handleStream(0xc0005b8a80, 0x553bb20, 0xc0005b9200, 0xc000533200, 0x0)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1245 +0xd25
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000365e0, 0xc0005b8a80, 0x553bb20, 0xc0005b9200, 0xc000533200)
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:685 +0x9f
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4: created by github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
2019-09-04T14:03:02.653+0100 [DEBUG] plugin.terraform-provider-aws_v2.25.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:683 +0xa1
2019-09-04T14:03:02.657+0100 [DEBUG] plugin: plugin process exited: path=/home/xxxx/.terraform/plugins/linux_amd64/terraform-provider-aws_v2.25.0_x4 pid=27167 error="exit status 2"
2019/09/04 14:03:02 [DEBUG] module.nginx-b.aws_instance.this[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
```

I'm assuming that this bug is related to the aws provider but the module shouldn't be setting the  `credit_specification` with nil. Maybe a better approach would be to use the `dynamic` and only set the `credit_specification` when the local != nil..

**Comment by dekimsey** [comment]
@nunofernandes I reported this in the AWS provider repository. The provider is panicking, there's no way that's not a bug. Have you identified a work-around in the meantime?

**Comment by dekimsey** [comment]
This is fixed in https://github.com/terraform-providers/terraform-provider-aws/issues/10203 and is expected to be released in AWS provider 2.30.0. I'm going to close my PR as its no longer relevant, but the work-around is there if someone needs it.

**Comment by antonbabenko** [comment]
Thanks @dekimsey for the update and fixing this in AWS provider. Closing this issue.
